### PR TITLE
iOS port Phase 5 core: ADIF export/import

### PR DESCRIPTION
**Pull request by patrickrb**
## What

Adds ADIF logging to the `FT8Engine` target, operating on the existing `QsoRecord` (from #287). No GRDB — pure string logic, fully host-testable. The persistence layer (`FT8Data`, GRDB) will call these later.

- **`Adif.export(_:)`** — byte-for-byte port of desktop `db.rs::export_adif`: the `FT8AF ADIF Export<eoh>` header, the `CALL` field + fixed `QSL_RCVD`/`QSL_MANUAL` flags, every non-empty optional field in the same order, then `comment` + `<eor>`. Field lengths use **UTF-8 byte count** (matching Rust `str::len`), so logs round-trip identically across the iOS app, the desktop port, and ShareLogs.
- **`Adif.parse(_:)`** — the inverse (new on iOS, since users import existing logs). A lenient length-prefixed `<field:len>value` tokenizer over the UTF-8 bytes, so a value may itself contain `<`/`>`; field names matched case-insensitively; header fields (before `<eoh>`) and unknown/QSL fields ignored; records end at `<eor>` (a trailing record without one is still accepted); an over-long length field is skipped rather than mis-read.

## Tests

`AdifTests`: exact desktop-format export, empty-optional omission, UTF-8 byte length; `export → parse` round trip; and parser edge cases — header/unknown/QSL skipping, angle-brackets-in-value (proves length-prefixed parsing), case-insensitivity, `:type` indicator, headerless + multiple records, trailing record without `<eor>`, malformed over-long field, and empty/header-only input.

## Scope

Decoder-independent, decision/format-only, targets `dev` directly. Lives in `FT8Engine` alongside `QsoRecord` to avoid a GRDB dependency now.

## Still draft

No Swift toolchain on Windows. On a Mac: `cd ios/FT8AFKit && swift test`. Ready once green.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: ios/FT8AFKit/Sources/FT8Engine/Adif.swift [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+            let parts = tag.split(separator: ":", omittingEmptySubsequences: false)
+            guard parts.count >= 2, let len = Int(parts[1]), len >= 0, i + len <= n else {
+                continue // skip a malformed/over-long field rather than mis-read
+            }
+            let name = parts[0].lowercased()
+            current[name] = String(decoding: bytes[i..<(i + len)], as: UTF8.self)
+            i += len
+        }
+        if !current.isEmpty { records.append(build(current)) } // trailing record w/o <eor>
+        return records
```
[the remark this answers is not in the document]
Comment (patrickrb) [reply ios/FT8AFKit/Sources/FT8Engine/Adif.swift:93]: Fixed in 5a45171e. Parsing now finalizes a record via `appendIfValid()`, which appends only when a non-empty `CALL` is present — so a record made up solely of ignored fields (QSL flags / unknown tags) is dropped instead of yielding an empty-call `QsoRecord`. Applies to both `<eor>` and the trailing-record path.

File: ios/FT8AFKit/Tests/FT8EngineTests/AdifTests.swift [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            "Generated by WSJT-X\n<adif_ver:5>3.1.0 <programid:6>WSJT-X <eoh>\n"
+            + "<call:5>K1ABC <qsl_rcvd:1>N <gridsquare:4>FN42 <mode:3>FT8 <foo:3>bar <eor>\n"
+        let parsed = Adif.parse(adif)
+        XCTAssertEqual(parsed.count, 1)
+        XCTAssertEqual(parsed[0].call, "K1ABC")
+        XCTAssertEqual(parsed[0].gridsquare, "FN42")
+        XCTAssertEqual(parsed[0].mode, "FT8")
+    }
+
+    func testParseValueContainingAngleBrackets() {
```
[the remark this answers is not in the document]
Comment (patrickrb) [reply ios/FT8AFKit/Tests/FT8EngineTests/AdifTests.swift:92]: Added in 5a45171e. `testParseDropsRecordsWithoutCall` feeds a CALL-less record (only QSL/unknown fields) followed by a valid one and asserts only the valid one survives; `testParseDropsEmptyCallRecord` covers an explicit empty `<call:0>`.
